geomesa-process/geomesa-process-vector/src/main/scala/org/locationtech/geomesa/process/transform/DateOffsetProcess.scala

```diff
@@ -0,0 +1,60 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.process.transform
+
+import java.time.{Period, ZoneId}
+import java.util.Date
+
+import org.geotools.data.collection.ListFeatureCollection
+import org.geotools.data.simple.SimpleFeatureCollection
+import org.geotools.feature.simple.SimpleFeatureImpl
+import org.geotools.process.ProcessException
+import org.geotools.process.factory.{DescribeParameter, DescribeProcess, DescribeResult}
+import org.locationtech.geomesa.features.ScalaSimpleFeature
+import org.locationtech.geomesa.process.GeoMesaProcess
+import org.locationtech.geomesa.utils.collection.CloseableIterator
+import org.opengis.feature.simple.SimpleFeature
+
+import scala.collection.JavaConversions._
+
+@DescribeProcess(
+  title = "Date Offset Process",
+  description = "Modifies the specified date field in a feature collection by an input time period."
+)
+class DateOffsetProcess extends GeoMesaProcess {
+
+  @throws(classOf[ProcessException])
+  @DescribeResult(name = "result", description = "Output collection")
+  def execute(@DescribeParameter(name = "data", description = "Input features")
+              obsFeatures: SimpleFeatureCollection,
+              @DescribeParameter(name = "dateField", description = "The date attribute to modify")
+              dateField: String,
+              @DescribeParameter(name = "timeOffset", description = "Time offset (e.g. P1D)")
```

Comment:
right now its only doing dates and not times...i bet this will fail because Period.parse() won't handle it.

```
"move a feature collection back in time one hour" in {  
      val collection = new ListFeatureCollection(sft, new Random(-1L).shuffle(features))
      val result = process.execute(collection, "dtg", "P-1H")
      val date = result.features().next.getAttribute("dtg").asInstanceOf[Date]
      val dt = new DateTime(date)
      dt.withSecondOfMinute(0) mustEqual new DateTime("2017-02-19T23:00:00.000Z")
    }
```

maybe we rename this datetimeoffsiteprocess or durationoffset or actually add time support. I think you may find that the java.time.Duration class can parse "combined" date times like "P5dT1H"

I think it may be nice to support offset by hours not just days.

Reply:
more info: https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#parse-java.lang.CharSequence-

Reply:
yes i know its weird i noticed that...but after you write a few ageoff iterators you learn iso8601 i guess...i should likely go test the ageoff itr too.